plugin/role_set.go

```diff
@@ -166,6 +169,32 @@ func (b *backend) saveRoleSetWithNewAccount(ctx context.Context, req *logical.Re
 		return nil, err
 	}
 
+	// Service accounts in GCP are eventually consistent, and can take over 60s
+	// to be ready for use. Attempt to GET the service account in a retry loop
+	// to ensure the service account is ready for use before continuing on
+	// and creating IAM bindings.
+	_, err = retryWithExponentialBackoff(ctx, func() (interface{}, bool, error) {
+		iamAdmin, err := b.IAMAdminClient(req.Storage)
+		if err != nil {
+			return nil, false, err
+		}
+		gcpAcct, err := b.getServiceAccount(iamAdmin, &gcputil.ServiceAccountId{
```

Comment:
Are there other API calls that we should retry?

Reply:
This looks like it will solve our issue. Is there a docker image I can point to for testing/validation?

Reply:
@TJM This was released yesterday as [v0.21.2](https://github.com/hashicorp/vault-plugin-secrets-gcp/releases/tag/v0.21.2). Please let us know if this solves your issue!

- https://hub.docker.com/r/hashicorp/vault-plugin-secrets-gcp/tags
- https://gallery.ecr.aws/hashicorp/vault-plugin-secrets-gcp